### PR TITLE
[SYCL][ESIMD][EMU] Checking header files required for CM_EMU build

### PR DESCRIPTION
**Pull request by dongkyunahn-intel**
This PR is for checking header files required for CM_EMU build. If check fails, configuration process terminates with messages for follow-up.

### CODE REVIEW

**Comment by kbobrovs** [comment]
Per discussion with @pvchupin, let's disable building emulator by default, instead add a flag `--enable-esimd-emulator` which will be used by CI to enable it during precommits, similarly to how CUDA part is tested.

**Comment by kbobrovs** [comment]
... but this check in the CMake files should remain and get exercised when emulator build is enabled

File: sycl/CMakeLists.txt [hunk 1]
```diff
@@ -336,11 +336,37 @@ if(SYCL_BUILD_PI_HIP)
...
+  if (NOT MSVC)
+    message(STATUS "Checking for header files required for CM_EMU library building - esimd_emulator support")
+    include(CheckIncludeFiles)
+    macro(CHECK_CM_EMU_REQUIRED_HEADER INCLUDE_FILE HAVE_FILE)
+      LIST (APPEND CMAKE_REQUIRED_INCLUDES "/usr/include/va")
+      LIST (APPEND CMAKE_REQUIRED_INCLUDES "/usr/include/x86_64-linux-gnu")
+      CHECK_INCLUDE_FILES(${INCLUDE_FILE} ${HAVE_FILE})
+      if (NOT ${HAVE_FILE})
+        UNSET(${HAVE_FILE} CACHE)
+        message(FATAL_ERROR
```
Comment (vladimirlaz) [inline sycl/CMakeLists.txt:349]: Can the message contain a missing file to let the developer fix the issue?

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

**Comment by dongkyunahn-intel** [comment]
LibVA dependency will be removed in CM_EMU. Closing this PR.

**Pull request closed without merging**